src/box/box.cc

```diff
@@ -4668,6 +4712,12 @@ box_read_ffi_enable(void)
 		box_read_ffi_is_disabled = false;
 }
 
+static void
+on_gc_garbage_collection(void)
```

Comment:
I did it this way because I didn't want gc.c to depend on box.h.
The same is done for wal_init, see on_wal_garbage_collection() and on_wal_checkpoint_threshold(), for example.

Reply:
This can be done without including box.h in gc.c. Add a function like gc_add_trigger(void (*cb)(void)) to gc.c and gc.h, that way it would be possible to register new callbacks for gc, without including extra headers in gc.c itself. Up to you though.